openmdao/vectors/vector.py

```diff
@@ -538,20 +548,26 @@ def set_var(self, name, val, idxs=_full_slice, flat=False):
 
         if flat:
             if isinstance(val, float):
-                self._views_flat[abs_name][idxs()] = val
+                self._views_flat[abs_name][idxs.flat()] = val
             else:
                 self._views_flat[abs_name][idxs.flat()] = np.asarray(val).flat
         else:
             value = np.asarray(val)
+            view = self._views[abs_name]
             try:
-                self._views[abs_name][idxs()] = value
+                if view.shape:
+                    view[idxs()] = value
+                else:
+                    # view is a scalar (so not really a view), so set the value into the
+                    # array using the flat view (which is actually a view)
+                    self._views_flat[abs_name][0] = value
             except Exception as err:
                 try:
-                    value = value.reshape(self._views[abs_name][idxs()].shape)
+                    value = value.reshape(view[idxs()].shape)
                 except Exception:
                     raise ValueError(f"{self._system().msginfo}: Failed to set value of "
                                      f"'{name}': {str(err)}.")
-                self._views[abs_name][idxs()] = value
+                view[idxs()] = value
 
     def dot(self, vec):
```

Comment:
Normally I'd say we need an example for docs, but in this case, the function API doesn't have any practical use until the FuncComp is added.

Reply:
Yeah there isn't much reason for someone to use the function API until we have function comps, and even if they do want to play with it, they could look at POEM 056 which explains it pretty thoroughly.